pkg/apiserver/statement/statement.go

```diff
@@ -210,3 +228,118 @@ func (s *Service) getPlanDetailHandler(c *gin.Context) {
 	}
 	c.JSON(http.StatusOK, result)
 }
+
+// @Router /statements/download/acquire_token [get]
+// @Summary Generate a download token for downloading statements
+// @Produce plain
+// @Param q query GetStatementsRequest true "Query"
+// @Success 200 {string} string "xxx"
+// @Security JwtAuth
+// @Failure 401 {object} utils.APIError "Unauthorized failure"
+func (s *Service) downloadTokenHandler(c *gin.Context) {
+	var req GetStatementsRequest
+	if err := c.ShouldBindQuery(&req); err != nil {
+		utils.MakeInvalidRequestErrorFromError(c, err)
+		return
+	}
+	db := utils.GetTiDBConnection(c)
+	fields := []string{}
+	if strings.TrimSpace(req.Fields) != "" {
+		fields = strings.Split(req.Fields, ",")
+	}
+	overviews, err := QueryStatementsOverview(
+		db,
+		req.BeginTime, req.EndTime,
+		req.Schemas,
+		req.StmtTypes,
+		req.Text,
+		fields)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	if len(overviews) == 0 {
+		utils.MakeInvalidRequestErrorFromError(c, errors.New("no data to export"))
+		return
+	}
+
+	// convert data
+	fieldsMap := make(map[string]string)
+	t := reflect.TypeOf(overviews[0])
+	fieldsNum := t.NumField()
+	for i := 0; i < fieldsNum; i++ {
+		field := t.Field(i)
+		fieldsMap[strings.ToLower(field.Tag.Get("json"))] = field.Name
+	}
+
+	csvData := [][]string{fields}
+	for _, overview := range overviews {
+		row := []string{}
+		v := reflect.ValueOf(overview)
+		for _, field := range fields {
+			filedName := fieldsMap[field]
+			s := v.FieldByName(filedName)
+			var val string
+			switch s.Interface().(type) {
+			case int:
+				val = strconv.FormatInt(s.Int(), 10)
+			default:
+				val = s.String()
+			}
+			row = append(row, val)
+		}
+		csvData = append(csvData, row)
+	}
+
+	// generate csv
+	timeLayout := "01021504"
+	beginTime := time.Unix(int64(req.BeginTime), 0).Format(timeLayout)
+	endTime := time.Unix(int64(req.EndTime), 0).Format(timeLayout)
+	tmpfile, err := ioutil.TempFile("", fmt.Sprintf("statements_%s_%s_*.csv", beginTime, endTime))
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	defer tmpfile.Close()
+	csvwriter := csv.NewWriter(tmpfile)
```

Comment:
In v5.0 we have a new security requirement: anything on the disk should not leak user data. Statements contain user data, so that maybe you can follow this way: the data is stream-encrypted on the disk using a non-persisted key and stream-decrypted and served to user when later a download token is used.

Reply:
Got!

Reply:
Done.